src/EventStore.Projections.Core/Services/Processing/EventReaderCoreService.cs

```diff
@@ -243,7 +245,17 @@ public class EventReaderCoreService :
 			_publisher.Publish(new EventReaderSubscriptionMessage.Failed(subscription.SubscriptionId, message.Reason));
 		}
 
-		
+		public void Handle(ReaderSubscriptionMessage.ReportProgress message) {
+			if (_stopped)
+				return;
+
+			foreach (var subscription in _subscriptions.Values) {
+				subscription.Handle(message);
+			}
+
+			_publisher.Publish(TimerMessage.Schedule.Create(TimeSpan.FromMilliseconds(500), new SendToThisEnvelope(this), new ReaderSubscriptionMessage.ReportProgress()));
```

Comment:
See above comment. These messages should share the same correlation id

Reply:
Also, should we be creating a new envelope every time?

Reply:
Done